batcher/aligned-sdk/src/sdk.rs

```diff
@@ -228,24 +430,115 @@ async fn receive(
     Ok(None)
 }
 
-fn verify_response(
-    verification_data_commitment: &VerificationDataCommitment,
-    batch_inclusion_data: &BatchInclusionData,
-) -> bool {
-    debug!("Verifying response data matches sent proof data ...");
-    let batch_inclusion_proof = batch_inclusion_data.batch_inclusion_proof.clone();
+async fn receive_and_wait(
```

Comment:
This function is too long, it should prob be refactored into different functions internally